samples/dashboards/demo/climate/demo.js

```diff
@@ -977,20 +976,29 @@ async function updateBoard(board, city, column, scale, newData) {
             data: cityTable.modified
                 .getRows(void 0, void 0, ['time', column])
         });
+
+        // Update range selection
+        selectionTable.setColumns(cityTable.modified.getColumns(), 0);
+        await selectionTable.setModifier();
     }
 
-    // Update range selection
-    selectionTable.setColumns(cityTable.modified.getColumns(), 0);
     const timeRangeMax = timeRangeSelector.chart.axes[0].max;
     const timeRangeMin = timeRangeSelector.chart.axes[0].min;
-    await selectionTable.setModifier(new RangeModifier({
-        modifier: 'Range',
-        ranges: [{
+    const selectionModifier = selectionTable.getModifier();
+
+    if (
+        !selectionModifier ||
+        selectionModifier.options.ranges[0].maxValue !== timeRangeMax ||
```

Comment:
Shouldn't we check that `selectionModifier.options` exist?

Reply:
`DataModifier.options` always exists. `ranges` is a default. The only thing that could fail is accessing index 0, but we know that we have set it.